build_tools/benchmarks/set_adreno_gpu_scaling_policy.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
+
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Runs on a phone with Adreno GPU to set the GPU frequency scaling policy.
+
+################################### WARNING ####################################
+# This will overheat the phone if it's not on a cooling plate, resulting in    #
+# thermal throttling. To prevent anything catching on fire, the actual GPU     #
+# frequencies will be throttled to below the maximum, skewing your results.    #
+################################################################################
+
+set -euo pipefail
+
+POLICY="${1:-performance}"
+
+readonly ADRENO_GPU_PATH="/sys/class/kgsl/kgsl-3d0"
+
+# Available frequencies are sorted, either in ascending or descending order.
+ADRENO_MAX_FREQ=$(cat "${ADRENO_GPU_PATH}/devfreq/available_frequencies" | tr " " "\n" | head -1)
+ADRENO_MIN_FREQ=$(cat "${ADRENO_GPU_PATH}/devfreq/available_frequencies" | tr " " "\n" | tail -1)
+if (( ${ADRENO_MAX_FREQ} < ${ADRENO_MIN_FREQ} )); then
+  TMP=${ADRENO_MAX_FREQ}
+  ADRENO_MAX_FREQ=${ADRENO_MIN_FREQ}
+  ADRENO_MIN_FREQ=${TMP}
+fi
+
+# Power levels match available freqencies.
+readonly ADRENO_MAX_PWRLEVEL=0
+(( ADRENO_MIN_PWRLEVEL = $(cat "${ADRENO_GPU_PATH}/num_pwrlevels") - 1 ))
```

Comment:
You can add a readonly declaration here afterward:

```suggestion
(( ADRENO_MIN_PWRLEVEL = $(cat "${ADRENO_GPU_PATH}/num_pwrlevels") - 1 ))
readonly ADRENO_MIN_PWRLEVEL
```